### PR TITLE
Casper Livenet

### PR DESCRIPTION
**Pull request by zie1ony**

### CODE REVIEW

File: examples/Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,12 @@ odra = { path = "../core", default-features = false }
 odra-modules = { path = "../modules", default-features = false }
 
 [features]
-default = ["mock-vm"]
+default = ["casper"]
```
Comment (kubaplas) [inline examples/Cargo.toml]: Why the default is casper?

File: examples/src/erc20.rs [hunk 2]
```diff
@@ -163,7 +163,7 @@ pub mod tests {
     }
 
     #[test]
-    fn initialization() {
+    fn initialization2() {
```
Comment (kubaplas) [inline examples/src/erc20.rs]: ?

File: lang/codegen/src/generator/module_impl/deploy.rs [hunk 3]
```diff
@@ -16,247 +20,73 @@ as_ref_for_contract_impl_generator!(Deploy);
...
+        .map(|arg| {
+            let pat = &*arg.pat;
+            quote!(args.get(stringify!(#pat)))
         })
-        .collect::<TokenStream>()
+        .collect::<Punctuated<TokenStream, Comma>>()
 }
 
-fn args_to_fn_args<'a, T>(args: T) -> Punctuated<TokenStream, Comma>
+fn args_to_fn_args2<'a, T>(args: T) -> Punctuated<TokenStream, Comma>
```
Comment (kubaplas) [inline lang/codegen/src/generator/module_impl/deploy.rs]: 2?

File: mock-vm/backend/src/storage.rs [hunk 4]
```diff
@@ -50,6 +50,7 @@ impl Storage {
     ) -> Result<Option<T>, MockVMSerializationError> {
         let hash = Storage::hashed_key(address, key);
         let result = self.state.get(&hash).cloned().map(|bytes| T::deser(bytes));
+        println!("{:?}", result.is_none());
```
Comment (kubaplas) [inline mock-vm/backend/src/storage.rs]: Remove

File: odra-casper/livenet/src/casper_client.rs [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+        serde_json::from_value(response.clone()).unwrap()
+    }
+}
+
+fn find_wasm_file_path(wasm_file_name: &str) -> PathBuf {
+    let mut path = PathBuf::from("wasm").join(wasm_file_name);
+    let mut checked_paths = vec![];
+    for _ in 0..2 {
+        if path.exists() {
+            println!("Found wasm under {:?}", path);
```
Comment (kubaplas) [inline odra-casper/livenet/src/casper_client.rs]: Maybe we should use logger like in cargo odra for such messages?

File: odra-casper/livenet/src/casper_client.rs [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+    use casper_types::{
+        bytesrepr::{FromBytes, ToBytes},
+        U256
+    };
+    use odra_casper_types::Address;
+
+    use super::CasperClient;
+
+    const CONTRACT_PACKAGE_HASH: &str =
+        "hash-40dd2fef4e994d2b0d3d415ce515446d7a1e389d2e6fc7c51319a70acf6f42d0";
```
Comment (kubaplas) [inline odra-casper/livenet/src/casper_client.rs:427]: Those values will be different depending on ENV_CHAIN_NAME, they should be also put in the env file, or the node and chain name also hardcoded for testing.

**Review by kubaplas — APPROVED** [review]

**Comment by kubaplas** [comment]
Add .env.sample file

File: odra-casper/livenet/src/casper_client.rs (hunk 5 above)
Comment (zie1ony) [reply odra-casper/livenet/src/casper_client.rs]: done

File: lang/codegen/src/generator/module_impl/deploy.rs (hunk 3 above)
Comment (zie1ony) [reply lang/codegen/src/generator/module_impl/deploy.rs]: fixed

**Pull request merged**